src/transformers/integrations/ggml.py

```diff
@@ -580,6 +627,6 @@ def convert_gguf_tokenizer(tokenizer_dict) -> Tokenizer:
         A instance of [`~tokenizers.Tokenizer`] to be used as the backend tokenizer of a
         [`~tokenization_utils_base.PreTrainedTokenizerFast`]
     """
-    tokenizer_class_name = tokenizer_dict["tokenizer_type"]
+    tokenizer_class_name = architecture
```

Comment:
can't we keep the previous logic?

Reply:
According to [gguf.md](https://github.com/ggerganov/ggml/blob/master/docs/gguf.md), most of tokenizers are converted to `llama`/`gpt2` format, which means we can only convert the gguf tokenizer to `LlamaTokenizer` or `GPT2Tokenizer`.

It seems that using a `GPT2Tokenizer` converted from gguf also works for qwen2 generally.  But I think it would be better to use architecture here so that we can customize the tokenizer convert for different situations to return the expected tokenizer.